### DISCUSSION TITLE
Impossible to install @casl/angular with angular 13

### DISCUSSION THREAD

**Discussion by Chachuke** [stalniy/casl#690] [Q&A]
Hi,

So i've got an ionic-angular project where i need to use casl.

I see on this page that i have to install the 7.x version of casl/angular
https://github.com/stalniy/casl/blob/master/packages/casl-angular/CHANGELOG.md

So, I tried, to install casl but unfortunatly i can't.

here is what i got as error : 
`npm install @casl/angular@7.0.0
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: xxxxx@0.0.1
npm ERR! Found: rxjs@6.6.7
npm ERR! node_modules/rxjs
npm ERR!   rxjs@"^6.6.7" from the root project
npm ERR!   peer rxjs@"^6.5.3 || ^7.4.0" from @angular/core@13.3.11
npm ERR!   node_modules/@angular/core
npm ERR!     @angular/core@"^13.3.11" from the root project
npm ERR!     peer @angular/core@"^13.0.0" from @casl/angular@7.0.0
npm ERR!     node_modules/@casl/angular
npm ERR!       @casl/angular@"7.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer rxjs@"^7.5.5" from @casl/angular@7.0.0
npm ERR! node_modules/@casl/angular
npm ERR!   @casl/angular@"7.0.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/xxx/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/xxx/.npm/_logs/2022-10-07T14_32_45_808Z-debug-0.log`

Here is my ionic info : 

![Capture d’écran 2022-10-07 à 16 53 48](https://user-images.githubusercontent.com/8606411/194583225-5aa52cb7-b564-455b-bc21-9dfbce3f741d.png)

So here i understand that @casl/angular need rxjs in version 7.5.5 and angular in version 13.0.0, the problem is that i can't update to version 7 of rxjs nor to version 14 of angular because of some dependencies used in the project...

is there something i can do to install it without changing angular or rxjs version ? 

I managed to install it by downgrading to angular 12, but that's not what i want.

**Comment by stalniy** [comment]
I see, angular 13 supports both versions `    "rxjs": "^6.5.3 || ^7.4.0"` but in casl only 1 specified. If it's still an issue, I'd recommend to ignore this (use `--force` or `overrides` in package.json). Since casl doesn't use anything fancy from rxjs
